platform/api/azure/storage_mit.go

```diff
@@ -290,8 +299,11 @@ func (a *API) SignBlob(storageaccount, storagekey, container, blob string) (stri
 	}
 
 	bsc := sc.GetBlobService()
-
-	return bsc.GetBlobSASURI(container, blob, time.Date(2099, time.December, 31, 23, 59, 59, 0, time.UTC), "rl") // TODO: Migrate to new API version
+	b := bsc.GetContainerReference(container).GetBlobReference(blob)
+	blobSASOptions := storage.BlobSASOptions{}
+	blobSASOptions.Read = true
+	blobSASOptions.Expiry = time.Date(2099, time.December, 31, 23, 59, 59, 0, time.UTC)
+	return b.GetSASURI(blobSASOptions) // TODO: Migrate to new API version
```

Comment:
Is this TODO still valid?

Reply:
Funny that you mention it. I was just looking into the Azure marketplace with Sayan and I realized this is wrong. The SAS token needs to be at container level (with read+list permissions) but the URI needs to refer to the blob. So this will get even uglier.

Reply:
Pushed a new version of this operation in the latest commit. I don't see a nicer way to do what the marketplace requires :(